### PR TITLE
Implement rooms list/search

### PR DESCRIPTION
**Pull request by Mehuge**
For review and feedback (don't merge yet)

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-patcher-ui/pull/69" target="_blank">ReviewNinja

File: src/ts/content/chat/Info.tsx [hunk 1]
```diff
@@ -49,7 +53,10 @@ class Info extends React.Component<InfoProps, InfoState> {
             leave={this.props.leaveRoom}
           />
         );
-        content.push(<JoinRoomButton key="join-button" join={this.props.selectRoom}/>);
+        content.push(
+          <JoinRoomButton key="join-button" join={this.props.selectRoom}
+            getRooms={this.getRooms}
```
Comment (saddieeiddas) [inline src/ts/content/chat/Info.tsx]: could this be single line?

File: src/ts/content/chat/JoinRoomButton.tsx [hunk 2]
```diff
@@ -17,32 +17,35 @@ export interface JoinRoomButtonState {
...
     this.props.join(new RoomId(room, chatType.GROUP));
-    this.setState({showJoinRoomModal: false});
+    this.setState({ showJoinRoomModal: false } as any);
   }
 
   generateJoinRoomModal = () => {
     return (
       <div className='fullscreen-blackout' key='join-room'>
-        <JoinRoomModal closeModal={this.closeModal} joinRoom={this.joinRoom} />
+        <JoinRoomModal closeModal={this.closeModal} joinRoom={this.joinRoom}
```
Comment (saddieeiddas) [inline src/ts/content/chat/JoinRoomButton.tsx]: could this be single line?

File: src/ts/content/chat/JoinRoomButton.tsx [hunk 3]
```diff
@@ -17,32 +17,35 @@ export interface JoinRoomButtonState {
...
     super(props);
     this.promptRoom = this.promptRoom.bind(this);
-    this.state = {showJoinRoomModal: false};
+    this.state = { showJoinRoomModal: false };
   }
 
   showModal = () => {
-    this.setState({showJoinRoomModal: true});
+    this.setState({ showJoinRoomModal: true } as any);
+    this.props.getRooms();        // temp
```
Comment (saddieeiddas) [inline src/ts/content/chat/JoinRoomButton.tsx]: should this be left in for now?

File: src/ts/chat/CSEChat.ts [hunk 4]
```diff
@@ -234,10 +261,30 @@ class CSEChat  {
...
   }
+  
+  _gotRooms(id: string, stanza: Element) : void {
+    const items : Element[] = stanza.getChildren('item');
+    const info : any = this._msgs[id];
+    if (info) {
+      this._inFlight --;
+      delete this._msgs[id]; 
+      const rooms : Room[] = [];
+      for (let i : number = 0; i < items.length; i++) {
```
Comment (saddieeiddas) [inline src/ts/chat/CSEChat.ts]: would it be possible to do `items.forEach((item) => {});` here ?

File: src/ts/content/chat/Info.tsx (hunk 1 above)
Comment (Mehuge) [inline src/ts/content/chat/Info.tsx]: !fixed

File: src/ts/content/chat/JoinRoomButton.tsx (hunk 3 above)
Comment (Mehuge) [inline src/ts/content/chat/JoinRoomButton.tsx]: !fixed

File: src/ts/content/chat/JoinRoomButton.tsx (hunk 2 above)
Comment (Mehuge) [inline src/ts/content/chat/JoinRoomButton.tsx]: !fixed

File: src/ts/chat/CSEChat.ts (hunk 4 above)
Comment (Mehuge) [inline src/ts/chat/CSEChat.ts]: !fixed

**Pull request merged**